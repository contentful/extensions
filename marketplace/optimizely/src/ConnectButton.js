import React from 'react';

export default function ConnectButton() {
  return (
    <svg id="Layer_1" data-name="Layer 1" width="40" height="40" viewBox="40 40">
      <path
        d="M25.5,12.77a7.65,7.65,0,0,0-1.91,3.28,7.31,7.31,0,0,0-5.41,7l5.15-5.15c0,.44,0,.89.09,1.38l-5.1,5.15a7.44,7.44,0,0,0,.75,2.17l3.28-3.28a3.24,3.24,0,0,1,.89-2.4,4.7,4.7,0,0,1,.53-.44h0a7.24,7.24,0,0,0,4.3,4.3h0a3.05,3.05,0,0,1-2.8,1.46L22,29.55a7.44,7.44,0,0,0,2.17.75l5.19-5.15a6.89,6.89,0,0,0,1.38.13l-5.19,5.1a7.31,7.31,0,0,0,7-5.41,7.28,7.28,0,1,0-3.77-14.07A7,7,0,0,0,25.5,12.77Zm7.37,7.37a3.11,3.11,0,1,1,0-4.44,3.19,3.19,0,0,1,0,4.44Z"
        style={{fill: '#fff'}}
      />
      <path
        d="M23.28,25.33a3.76,3.76,0,0,1-.62-.84l-3.06,3,.67.8c.27.27.53.49.8.71l3-3Z"
        style={{fill: '#fff'}}
      />
    </svg>
  );
}
