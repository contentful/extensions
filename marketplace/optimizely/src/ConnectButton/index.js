import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import { Button } from '@contentful/forma-36-react-components';

import OptimizelyLogo from './OptimizelyLogo';

const styles = {
  connect: css({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    '&:hover': {
      backgroundColor: 'transparent!important' // necessary to eliminate the forma styling in favor of the custom optimizely styling
    }
  })
};

export default function ConnectButton({ openAuth }) {
  return (
    <Button
      className={styles.connect}
      onClick={openAuth}
      testId="connect-button"
      isFullWidth
      buttonType="naked">
      <OptimizelyLogo />
    </Button>
  );
}

ConnectButton.propTypes = {
  openAuth: PropTypes.func.isRequired
};
