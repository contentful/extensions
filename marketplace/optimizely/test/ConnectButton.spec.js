import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';

import ConnectButton from '../src/ConnectButton';

describe('ConnectButton', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = render(<ConnectButton openAuth={() => {}} />);

    expect(container).toMatchSnapshot();
  });

  it('should handle the openAuth functionality onClick', () => {
    const openMock = jest.fn();
    const { getByTestId } = render(<ConnectButton openAuth={openMock} />);

    fireEvent.click(getByTestId('connect-button'));

    expect(openMock).toHaveBeenCalledTimes(1);
  });
});
