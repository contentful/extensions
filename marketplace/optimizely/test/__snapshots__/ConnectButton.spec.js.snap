// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectButton should match snapshot 1`] = `
<div>
  <button
    class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-4k45fw Button__Button--naked___mB6LS Button__Button--full-width___3Fmpo"
    data-test-id="cf-ui-button"
    data-testid="connect-button"
    type="button"
  >
    <span
      class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
      tabindex="-1"
    >
      <span
        class="Button__Button__label___3tcOj"
      >
        <svg
          data-name="Layer 1"
          height="42"
          id="Layer_1"
          viewBox="0 0 216 42"
          width="216"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            ConnectWithOptimizelyButton
          </title>
          <rect
            height="38"
            rx="5"
            ry="5"
            style="fill: #4069ff;"
            width="213.5"
            x="1.5"
            y="1.84"
          />
          <path
            d="M45.05,21.34v-.1a4.47,4.47,0,0,1,4.51-4.75c2.13,0,3.78,1.07,4,3.25H51.8a2,2,0,0,0-2.22-1.87c-1.64,0-2.7,1.27-2.7,3.35v.1c0,2.09,1,3.33,2.72,3.33a2.17,2.17,0,0,0,2.38-2h1.69c-.27,2.24-1.86,3.37-4.07,3.37C46.68,26.06,45.05,24.2,45.05,21.34Z"
            style="fill: #fff;"
          />
          <path
            d="M54.57,22.6v-.1a3.49,3.49,0,1,1,7,0v.1a3.49,3.49,0,0,1-7,0Zm5.37,0v-.09c0-1.4-.7-2.26-1.87-2.26s-1.89.84-1.89,2.25v.1c0,1.39.69,2.26,1.89,2.26S59.94,24,59.94,22.57Z"
            style="fill: #fff;"
          />
          <path
            d="M63,19.13h1.57v1.08A2.34,2.34,0,0,1,66.74,19c1.37,0,2.33.79,2.33,2.67v4.28H67.49V21.79c0-1-.42-1.5-1.35-1.5a1.51,1.51,0,0,0-1.6,1.63v4H63Z"
            style="fill: #fff;"
          />
          <path
            d="M70.87,19.13h1.57v1.08A2.34,2.34,0,0,1,74.64,19C76,19,77,19.78,77,21.65v4.28H75.39V21.79c0-1-.42-1.5-1.35-1.5a1.51,1.51,0,0,0-1.6,1.63v4H70.87Z"
            style="fill: #fff;"
          />
          <path
            d="M78.32,22.6v-.1A3.31,3.31,0,0,1,81.7,19a3.07,3.07,0,0,1,3.21,3.43v.46h-5c.05,1.29.7,2,1.87,2,.95,0,1.44-.38,1.56-1h1.52c-.22,1.43-1.38,2.21-3.12,2.21A3.26,3.26,0,0,1,78.32,22.6Zm5-.78c-.08-1.17-.68-1.69-1.65-1.69A1.74,1.74,0,0,0,80,21.82Z"
            style="fill: #fff;"
          />
          <path
            d="M85.86,22.61v-.1A3.35,3.35,0,0,1,89.29,19a2.78,2.78,0,0,1,3.11,2.55H90.88a1.4,1.4,0,0,0-1.56-1.29c-1.08,0-1.85.83-1.85,2.24v.1c0,1.48.73,2.26,1.89,2.26A1.55,1.55,0,0,0,91,23.39h1.44a2.88,2.88,0,0,1-3.15,2.67A3.25,3.25,0,0,1,85.86,22.61Z"
            style="fill: #fff;"
          />
          <path
            d="M94,24V20.31h-.91V19.13H94v-1.5h1.57v1.5h1.49v1.18H95.62v3.58c0,.6.29.87.78.87a1.94,1.94,0,0,0,.77-.13v1.25A3,3,0,0,1,96.1,26,1.83,1.83,0,0,1,94,24Z"
            style="fill: #fff;"
          />
          <path
            d="M101,19.13h1.68L104,24l1.44-4.91h1.31L108.11,24l1.42-4.91h1.53l-2.13,6.8h-1.6l-1.26-4.58-1.38,4.58h-1.61Z"
            style="fill: #fff;"
          />
          <path
            d="M112.13,17.11a.94.94,0,1,1,.94.9A.91.91,0,0,1,112.13,17.11Zm.17,2h1.57v6.8H112.3Z"
            style="fill: #fff;"
          />
          <path
            d="M116,24V20.31h-.91V19.13H116v-1.5h1.57v1.5h1.49v1.18h-1.49v3.58c0,.6.29.87.78.87a1.94,1.94,0,0,0,.77-.13v1.25a3,3,0,0,1-1.07.17A1.83,1.83,0,0,1,116,24Z"
            style="fill: #fff;"
          />
          <path
            d="M120.58,16h1.57v4.23a2.34,2.34,0,0,1,2.2-1.22c1.37,0,2.33.79,2.33,2.67v4.28H125.1V21.79c0-1-.42-1.5-1.35-1.5a1.51,1.51,0,0,0-1.6,1.63v4h-1.57Z"
            style="fill: #fff;"
          />
          <path
            d="M131.24,21.34v-.1a4.56,4.56,0,0,1,4.71-4.73,4.5,4.5,0,0,1,4.68,4.69v.1a4.69,4.69,0,1,1-9.39,0Zm7.59,0v-.1c0-1.91-1.07-3.33-2.89-3.33s-2.9,1.34-2.9,3.34v.1c0,2,1.24,3.34,2.92,3.34S138.83,23.33,138.83,21.31Z"
            style="fill: #fff;"
          />
          <path
            d="M142.16,19.13h1.57v1.09A2.63,2.63,0,0,1,146,19c1.7,0,3,1.26,3,3.46v.1c0,2.18-1.26,3.51-3,3.51a2.46,2.46,0,0,1-2.22-1.2v3.39h-1.57Zm5.2,3.44v-.1c0-1.51-.78-2.25-1.81-2.25s-1.87.74-1.87,2.25v.1c0,1.52.74,2.24,1.88,2.24S147.36,24,147.36,22.57Z"
            style="fill: #fff;"
          />
          <path
            d="M150.65,24V20.31h-.91V19.13h.91v-1.5h1.57v1.5h1.49v1.18h-1.49v3.58c0,.6.29.87.78.87a1.94,1.94,0,0,0,.77-.13v1.25a3,3,0,0,1-1.07.17A1.83,1.83,0,0,1,150.65,24Z"
            style="fill: #fff;"
          />
          <path
            d="M155.06,17.11a.94.94,0,1,1,.94.9A.91.91,0,0,1,155.06,17.11Zm.17,2h1.57v6.8h-1.57Z"
            style="fill: #fff;"
          />
          <path
            d="M158.71,19.13h1.57v1A2.28,2.28,0,0,1,162.35,19a1.91,1.91,0,0,1,1.92,1.22,2.68,2.68,0,0,1,2.3-1.22c1.25,0,2.26.77,2.26,2.64v4.3h-1.57v-4.2c0-1-.43-1.43-1.21-1.43a1.45,1.45,0,0,0-1.5,1.56v4.07H163v-4.2c0-1-.44-1.43-1.21-1.43a1.45,1.45,0,0,0-1.5,1.56v4.07h-1.57Z"
            style="fill: #fff;"
          />
          <path
            d="M170.51,17.11a.94.94,0,1,1,.94.9A.91.91,0,0,1,170.51,17.11Zm.17,2h1.57v6.8h-1.57Z"
            style="fill: #fff;"
          />
          <path
            d="M173.45,25.75l3.6-5.37H173.8V19.13h5.58v.22l-3.59,5.32h3.48v1.26h-5.82Z"
            style="fill: #fff;"
          />
          <path
            d="M180.06,22.6v-.1A3.31,3.31,0,0,1,183.44,19a3.07,3.07,0,0,1,3.21,3.43v.46h-5c.05,1.29.7,2,1.87,2,.95,0,1.44-.38,1.56-1h1.52c-.22,1.43-1.38,2.21-3.12,2.21A3.26,3.26,0,0,1,180.06,22.6Zm5-.78c-.08-1.17-.68-1.69-1.65-1.69a1.74,1.74,0,0,0-1.74,1.69Z"
            style="fill: #fff;"
          />
          <path
            d="M188.09,16h1.57v9.95h-1.57Z"
            style="fill: #fff;"
          />
          <path
            d="M193.56,25.24l-2.69-6.11h1.7l1.81,4.38L196,19.13h1.53l-3.71,9.13h-1.55Z"
            style="fill: #fff;"
          />
          <path
            d="M25.5,12.77a7.65,7.65,0,0,0-1.91,3.28,7.31,7.31,0,0,0-5.41,7l5.15-5.15c0,.44,0,.89.09,1.38l-5.1,5.15a7.44,7.44,0,0,0,.75,2.17l3.28-3.28a3.24,3.24,0,0,1,.89-2.4,4.7,4.7,0,0,1,.53-.44h0a7.24,7.24,0,0,0,4.3,4.3h0a3.05,3.05,0,0,1-2.8,1.46L22,29.55a7.44,7.44,0,0,0,2.17.75l5.19-5.15a6.89,6.89,0,0,0,1.38.13l-5.19,5.1a7.31,7.31,0,0,0,7-5.41,7.28,7.28,0,1,0-3.77-14.07A7,7,0,0,0,25.5,12.77Zm7.37,7.37a3.11,3.11,0,1,1,0-4.44,3.19,3.19,0,0,1,0,4.44Z"
            style="fill: #fff;"
          />
          <path
            d="M23.28,25.33a3.76,3.76,0,0,1-.62-.84l-3.06,3,.67.8c.27.27.53.49.8.71l3-3Z"
            style="fill: #fff;"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;
